protocols/virtualdefund/virtualdefund.go

```diff
@@ -55,7 +55,7 @@ type Objective struct {
 const ObjectivePrefix = "VirtualDefund-"
 
 // newObjective constructs a new virtual defund objective
-func newObjective(preApprove bool, vFixed state.FixedPart, initialOutcome outcome.SingleAssetExit, paidToBob *big.Int, myRole uint) Objective {
+func newObjective(preApprove bool, vFixed state.FixedPart, initialOutcome outcome.SingleAssetExit, paidToBob *big.Int, toMyLeft, toMyRight *consensus_channel.ConsensusChannel, myRole uint) Objective {
```

Comment:
Shouldn't we match the pattern here

https://github.com/statechannels/go-nitro/blob/ca82de1b3ddbf963b0fdb53ddad10633d29b613c/protocols/virtualfund/virtualfund.go#L131-L132

and pass a _getter_ instead of the actual ConsensusChannels?

Reply:
Currently this is just a private `constructor` used mostly for testing. I think we can kick this down the road and implement a proper constructor when we need to call it from the engine in https://github.com/statechannels/go-nitro/issues/564 ?